.github/workflows/update-docs-in-website.yml

```diff
@@ -0,0 +1,120 @@
+        run: |
+          git checkout -b update-bindings-docs-${{ github.sha }}
+      - name: Update edit-page-config.json
+        uses: actions/github-script@v4
+        with:
+          script: |
+            const { writeFile } = require('fs').promises;
+            const configPath = './website/config/edit-page-config.json';
+            const checkSlug = 'reference/bindings/';
+            const slug = {
+              "value": checkSlug,
+              "href": "https://github.com/asyncapi/bindings/tree/master"
+            };
+
+            const configData = require(configPath);
+            const entryExists = configData.some(entry => entry.value === checkSlug);
+            if (!entryExists) {
+              configData.unshift(slug);
+              await writeFile(configPath, JSON.stringify(configData, null, 2))
+            }
+
+      - name: Update title and weight of the markdown files
+        uses: actions/github-script@v4
+        with:
+          script: |
+            const fs = require('fs').promises;
+            const path = require('path');
+            const rootPath = './bindings/';
+            let itemIndex = 10;
+
+            async function processMarkdownFiles(folderPath, isRoot = true) {
+              const items = await fs.readdir(folderPath, { withFileTypes: true });
+              for (const item of items) {
+                const fullPath = path.join(folderPath, item.name);
+                if (item.isDirectory()) {
+                  // Always process subdirectories, mark isRoot as false for recursive calls
+                  await processMarkdownFiles(fullPath, false);
+                } else if (item.name.endsWith('.md') && !isRoot) { // Skip root level .md files
+                  const baseName = path.basename(fullPath, '.md');
+                  const parentDirName = path.basename(folderPath);
+                  const newFileName = `${parentDirName}.md`;
+                  const newFullPath = path.join(folderPath, newFileName);
+                  await fs.rename(fullPath, newFullPath);
+                  
+                  const newData = `---\ntitle: '${parentDirName.charAt(0).toUpperCase() + parentDirName.slice(1)}'\nweight: ${itemIndex}\n---\n\n`;
+                  let existingFileData = await fs.readFile(newFullPath, 'utf8');
+                  
+                  existingFileData = existingFileData.replace(/<img\s+src="(?!http)(.*?)"/g, (match, src) => {
+                    // Remove './' prefix from src path and prepend '/img/docs/'
+                    const updatedSrc = src.replace(/^\.\//, '');
+                    return `<img src="/img/docs/${updatedSrc}"`;
+                    });
+                  
+                  const updatedContent = newData + existingFileData;
+                  await fs.writeFile(newFullPath, updatedContent);
+                  itemIndex++;
+                }
+              }
+            }
+
+            await processMarkdownFiles(rootPath);
```

Comment:
what is the reason for wrapping the code in async function that you later trigger anyway?